queue/abstract.lua

```diff
@@ -570,15 +570,43 @@ function method.start()
     end
 
     for _, tube_tuple in _queue:pairs() do
-        local tube = recreate_tube(tube_tuple)
-        -- gh-66: release all taken tasks on start
-        tube_release_all_tasks(tube)
+        -- Recreate tubes for registered drivers only.
+        if queue.driver[tube_tuple[4]] ~= nil then
```

Comment:
For my taste it would be better to assign `tube_tuple[4]` to a separate variable for clearness.
Although it's  up to you keep it as is or apply a patch.

```
--- a/queue/abstract.lua
+++ b/queue/abstract.lua
@@ -571,7 +571,8 @@ function method.start()
 
     for _, tube_tuple in _queue:pairs() do
         -- Recreate tubes for registered drivers only.
-        if queue.driver[tube_tuple[4]] ~= nil then
+        local tube_type = tube_tuple[4]
+        if queue.driver[tube_type] ~= nil then
             local tube = recreate_tube(tube_tuple)
             -- gh-66: release all taken tasks on start
             tube_release_all_tasks(tube)
```

Reply:
I understand you pain. I think the best way is to do as in fifottl driver - to use something like enum. I think I will send a similar patch later. But I don't want to create an additional variable for this. As a compromise, I've added a comment.